azuread/resource_application.go

```diff
@@ -300,6 +353,18 @@ func resourceApplicationUpdate(d *schema.ResourceData, meta interface{}) error {
 		properties.RequiredResourceAccess = expandADApplicationRequiredResourceAccess(d)
 	}
 
+	if d.HasChange("app_role") {
+		var disableAppRoles = true
```

Comment:
I'm not sure the logic here, could you explain it in a comment?

Reply:
this has been removed and the logic improved also with comments